src/platform/imx8/lib/clk.c

```diff
@@ -3,26 +3,62 @@
 // Copyright(c) 2019 Intel Corporation. All rights reserved.
 //
 // Author: Tomasz Lauda <tomasz.lauda@linux.intel.com>
+//         Janusz Jankowski <janusz.jankowski@linux.intel.com>
 
 #include <sof/common.h>
+#include <sof/drivers/ssp.h>
 #include <sof/lib/clk.h>
+#include <sof/lib/notifier.h>
 
 static struct freq_table platform_cpu_freq[] = {
-	{ 666000000, 666000, 0x0 }, /* default */
+	{ 666000000, 666000 },
 };
 
 STATIC_ASSERT(NUM_CPU_FREQ == ARRAY_SIZE(platform_cpu_freq),
 	      invalid_number_of_cpu_frequencies);
 
-struct freq_table *cpu_freq = platform_cpu_freq;
-
 /* TODO: abstract SSP clk based on platform and remove this from here! */
 static struct freq_table platform_ssp_freq[] = {
-	{ 19200000, 19200, 19 }, /* default */
```

Comment:
Is it not possible to completely remove this table? i.MX doesn't have SSP.

Reply:
```
diff --git a/src/platform/imx8/lib/clk.c b/src/platform/imx8/lib/clk.c
index 07169d3e..e9e5846b 100644
--- a/src/platform/imx8/lib/clk.c
+++ b/src/platform/imx8/lib/clk.c
@@ -17,23 +17,6 @@ static struct freq_table platform_cpu_freq[] = {
 STATIC_ASSERT(NUM_CPU_FREQ == ARRAY_SIZE(platform_cpu_freq),
              invalid_number_of_cpu_frequencies);
 
-/* TODO: abstract SSP clk based on platform and remove this from here! */
-static struct freq_table platform_ssp_freq[] = {
-       { 19200000, 19200 },
-       { 25000000, 25000 },
-};
-
-uint32_t platform_ssp_freq_sources[] = {
-       19,
-       12,
-};
-
-STATIC_ASSERT(NUM_SSP_FREQ == ARRAY_SIZE(platform_ssp_freq),
-             invalid_number_of_ssp_frequencies);
-
-struct freq_table *ssp_freq = platform_ssp_freq;
-uint32_t *ssp_freq_sources = platform_ssp_freq_sources;
-
 static struct clock_info platform_clocks_info[NUM_CLOCKS];
 
 struct clock_info *clocks = platform_clocks_info;
@@ -52,13 +35,4 @@ void platform_clock_init(void)
                        .set_freq = NULL,
                };
        }
-
-       platform_clocks_info[CLK_SSP] = (struct clock_info) {
-               .freqs_num = NUM_SSP_FREQ,
-               .freqs = platform_ssp_freq,
-               .default_freq_idx = SSP_DEFAULT_IDX,
-               .notification_id = NOTIFIER_ID_SSP_FREQ,
-               .notification_mask = NOTIFIER_TARGET_CORE_ALL_MASK,
-               .set_freq = NULL,
-       };
 }
```

You can also add the above patch to remove the i.MX specific part of SSP. A local test on the old branch (PLEASE rebase on top of master, I have the capture there) tells me it still seems to work, but I still want to be sure it works after rebase.

Reply:
Oh, also remember to revert 8e16ced80c (it's no longer needed now). We must test that too.

Reply:
@paulstelian97 Sorry for late reply, was ooo. I was just changing interface and adjusting whatever was there, I don't know what was stub and what was really physically on IMX platforms, feel free to remove these tables if they are not needed at all.

Reply:
@jajanusz I have removed them in the meanwhile actually AND fixed a bug that I introduced during that removal. It's fine :)